From: Tzafrir Cohen <tzafrir.cohen@xorcom.com>
Date: Sat, 12 Nov 2016 20:15:12 +0200
Origin: https://gerrit.asterisk.org/4404
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-26540
Subject: Add support for building RADUIS with radcli

Radcli is yet another RADIUS client library, generally compatible with
freeradius and radiusclient-ng.

This commit adds autoconf option for detecting it as well and changes
cdr_radius and cel_radius to use its header file in that case.

ASTERISK-26540 #close

Index: trunk/cdr/cdr_radius.c
===================================================================
--- trunk.orig/cdr/cdr_radius.c
+++ trunk/cdr/cdr_radius.c
@@ -41,11 +41,7 @@
 
 ASTERISK_FILE_VERSION(__FILE__, "$Revision$")
 
-#ifdef FREERADIUS_CLIENT
-#include <freeradius-client.h>
-#else
-#include <radiusclient-ng.h>
-#endif
+#include RADIUS_HEADER_STR
 
 #include "asterisk/channel.h"
 #include "asterisk/cdr.h"
Index: trunk/cel/cel_radius.c
===================================================================
--- trunk.orig/cel/cel_radius.c
+++ trunk/cel/cel_radius.c
@@ -35,11 +35,7 @@
 
 ASTERISK_FILE_VERSION(__FILE__, "$Rev$")
 
-#ifdef FREERADIUS_CLIENT
-#include <freeradius-client.h>
-#else
-#include <radiusclient-ng.h>
-#endif
+#include RADIUS_HEADER_STR
 
 #include "asterisk/channel.h"
 #include "asterisk/cel.h"
Index: trunk/configure.ac
===================================================================
--- trunk.orig/configure.ac
+++ trunk/configure.ac
@@ -2365,9 +2365,17 @@ AST_EXT_LIB_CHECK([LUA], [lua], [luaL_op
 # just different header filenames and different SONAMEs
 AST_EXT_LIB_CHECK([RADIUS], [freeradius-client], [rc_read_config], [freeradius-client.h])
 if test "x${PBX_RADIUS}" = "x1"; then
-	AC_DEFINE(FREERADIUS_CLIENT, [], [Use the FreeRADIUS-client library])
+	AC_DEFINE(RADIUS_HEADER_STR, [<freeradius-client.h>], [Name of RADIUS library include header])
 else
 	AST_EXT_LIB_CHECK([RADIUS], [radiusclient-ng], [rc_read_config], [radiusclient-ng.h])
+	if test "x${PBX_RADIUS}" = "x1"; then
+		AC_DEFINE(RADIUS_HEADER_STR, [<radiusclient-ng.h>], [Name of RADIUS library include header])
+	else
+		AST_EXT_LIB_CHECK([RADIUS], [radcli], [rc_read_config], [radcli/radcli.h])
+		if test "x${PBX_RADIUS}" = "x1"; then
+			AC_DEFINE(RADIUS_HEADER_STR, [<radcli/radcli.h>], [Name of RADIUS library include header])
+		fi
+	fi
 fi
 
 AST_EXT_LIB_CHECK([COROSYNC], [cpg], [cpg_join], [corosync/cpg.h], [-lcfg])
